elastic/kibana
# [APM] Create sub-feature role to manage APM settings write permissions

## Summary

Closes https://github.com/elastic/kibana/issues/156898

This PR adds a new sub-feature role to APM, which is the ability to write into the settings page: for UI, `settings:save`, for API, `apm_settings_write`.

The other changes are adapting to use this new permission.

## How to test
1. Go under `Stack Management` -> `Roles` and create a new custom role.
3. For Kibana, select `All spaces` for the space selector, and `Customize`, you can get all the permissions you need.
4. Go into `Observability` and `APM and User Experience`.
5. Select `Read` and save the role.
6. Create a new user and assign that role and also the `viewer` role.
7. Login with an incognito / different browser into the new user.
8. Go into `APM` -> `Settings`, WARNING: if you are not able to see settings is because you don't have data, run `node scripts/synthtrace many_services.ts --live --clean`.
9. You should not be able to change the configuration on each tab.
10. Change the role privileges to have `Read` but with write access.
11. Test it, you should be able to modify the settings.
12. Do the same with `All` with and without the write permissions.

## Demo

https://github.com/user-attachments/assets/5c1b6c33-4c2c-4616-bfe2-5b4bdc414e6a
